src/main/scala/au/com/agiledigital/toolform/model/PortMapping.scala

```diff
@@ -0,0 +1,72 @@
+package au.com.agiledigital.toolform.model
+
+import enumeratum.{Enum, EnumEntry}
+
+import scala.collection.immutable.IndexedSeq
+
+/**
+  * Defines a port mapping on a service.
+  *
+  * This is used to abstract away all the different syntax used by various backends.
+  *
+  * @param port           the port that is exposed by the service.
+  * @param containerPort  the port that is exposed by the container. This is optional and defaults to the same value as port.
+  * @param protocol       the protocol used by the port. This defaults to Tcp.
+  */
+case class PortMapping(port: Int, containerPort: Option[Int], protocol: PortProtocolType = PortProtocolType.Tcp) {
+
+  /**
+    * Returns the formatted string representation of this object that can be parsed back in again with the
+    * parsePortMappingFromConfigString function. It is compatible with Docker Compose port format.
+    *
+    * @see https://docs.docker.com/compose/compose-file/#short-syntax-1
+    *
+    * @return the string representation of this object.
+    */
+  override def toString: String = {
+    val protocolString = protocol.toString.toLowerCase
+    containerPort match {
+      case Some(containerPort) => s"$port:$containerPort/$protocolString"
+      case None                => s"$port/$protocolString"
+    }
+  }
+}
+
+object PortMapping {
+
+  private val parseRegex = """^(\d+)(?::(\d+))?(?:\/(udp|tcp))?$""".r
```

Comment:
`PortMappingRegex` see https://docs.scala-lang.org/style/naming-conventions.html#constants-values-variable-and-methods

Reply:
Fixed in c2dd744